pkg/ccl/logictestccl/testdata/logic_test/super_regions

```diff
@@ -0,0 +1,654 @@
+                                                 voter_constraints = '{+region=us-central-1: 2}',
+                                                 lease_preferences = '[[+region=us-central-1]]'
+
+query TT
+SHOW ZONE CONFIGURATION FOR PARTITION "ap-southeast-2" OF TABLE mr3.rt
+----
+PARTITION "ap-southeast-2" OF TABLE mr3.public.rt  ALTER PARTITION "ap-southeast-2" OF TABLE mr3.public.rt CONFIGURE ZONE USING
+                                                   range_min_bytes = 134217728,
+                                                   range_max_bytes = 536870912,
+                                                   gc.ttlseconds = 90000,
+                                                   num_replicas = 6,
+                                                   num_voters = 5,
+                                                   constraints = '{+region=ap-southeast-2: 2, +region=ca-central-1: 1, +region=us-central-1: 1, +region=us-east-1: 1, +region=us-west-1: 1}',
+                                                   voter_constraints = '{+region=ap-southeast-2: 2}',
+                                                   lease_preferences = '[[+region=ap-southeast-2]]'
+
+# a user should not be able to drop a region that is a member of a super region.
+statement error pq: region us-east-1 is part of super region test
+ALTER DATABASE db DROP REGION "us-east-1";
+
+# Dropping primary region.
+statement ok
+CREATE DATABASE db2 PRIMARY REGION "ca-central-1";
+ALTER DATABASE db2 ADD SUPER REGION "test" VALUES "ca-central-1";
+
+statement error pq: region ca-central-1 is part of super region test
+ALTER DATABASE db2 DROP REGION "ca-central-1";
+
+# Test the case where we have regional tables in the primary region. Initially
+# the primary region is part of a super region but we alter the primary region
+# to be part of another super region.
+statement ok
+CREATE DATABASE db3 PRIMARY REGION "ca-central-1" REGIONS "us-east-1", "us-west-1";
+ALTER DATABASE db3 ADD SUPER REGION "test1" VALUES "ca-central-1";
+ALTER DATABASE db3 ADD SUPER REGION "test2" VALUES "us-east-1";
+
+statement ok
+CREATE TABLE db3.t() LOCALITY REGIONAL BY TABLE;
+
+query TT
+SHOW ZONE CONFIGURATION FOR TABLE db3.t
+----
+TABLE db3.public.t  ALTER TABLE db3.public.t CONFIGURE ZONE USING
+                    range_min_bytes = 134217728,
+                    range_max_bytes = 536870912,
+                    gc.ttlseconds = 90000,
+                    num_replicas = 3,
+                    num_voters = 3,
+                    constraints = '{+region=ca-central-1: 1}',
+                    voter_constraints = '[+region=ca-central-1]',
+                    lease_preferences = '[[+region=ca-central-1]]'
+
+statement error pq: the current primary region is a member of a super region\n.*HINT: You can enable this operation by running `SET alter_primary_region_super_region_override = 'on'`
```

Comment:
This is one of the UX points I wanted to bring up, in this case, super region is a member of a primary region so we block here to prevent users from unknowingly moving all REGIONAL TABLES in the primary region (without an explicitly set region) to another super region.

@awoods187